### PR TITLE
docs: add hello world example with VPCConfig

### PR DESCRIPTION
**Pull request by heitorlessa**
*Issue #, if available:* Issue created in SAM CLI: https://github.com/awslabs/aws-sam-cli/issues/617

*Description of changes:*

Adds a hello world VPC example to demonstrate how to add correct permissions for Lambda VPC enabled using SAM Policy Templates including link to the docs for other examples.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by heitorlessa** [comment]
cc @brettstack

File: examples/2016-10-31/hello_world_vpc/template.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      Handler: index.handler
+      Runtime: nodejs6.10
+      CodeUri: src/
+      Policies:
+        - VPCAccessPolicy: {} 
+        # This policy gives permission for Lambdas to create/manage ENIs
+        # SAM Policy templates you can use: https://github.com/awslabs/serverless-application-model/blob/develop/examples/2016-10-31/policy_templates/all_policy_templates.yaml
+      VpcConfig:
+        SecurityGroupIds:
+          - String # add Security Group ID(s) for this Lambda
```
Comment (brettstack) [inline examples/2016-10-31/hello_world_vpc/template.yaml]: Would prefer the example to either create the SecurityGroupId(s) and SubnetId(s) or accept them as `Parameters` and `!Ref` them here.

File: examples/2016-10-31/hello_world_vpc/template.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+AWSTemplateFormatVersion : '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: A hello world application with VPC Access.
+
+Resources:
+  HelloWorldFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.handler
+      Runtime: nodejs6.10
```
Comment (brettstack) [inline examples/2016-10-31/hello_world_vpc/template.yaml]: nodejs8.10

File: examples/2016-10-31/hello_world_vpc/template.yaml (hunk 1 above)
Comment (heitorlessa) [reply examples/2016-10-31/hello_world_vpc/template.yaml]: Done

File: examples/2016-10-31/hello_world_vpc/template.yaml (hunk 2 above)
Comment (heitorlessa) [reply examples/2016-10-31/hello_world_vpc/template.yaml]: Done

File: examples/2016-10-31/hello_world_vpc/src/index.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+console.log('Loading function');
+
+exports.handler = (event, context, callback) => {
```
[the remark this answers is not in the document]
Comment (heitorlessa) [reply examples/2016-10-31/hello_world_vpc/src/index.js]: That's pretty cool!!!! Changed file following this format

File: examples/2016-10-31/hello_world_vpc/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+
+# Lambda function with VPC Access
+
+This example shows you how to create a Lambda function in VPC with the appropriate permissions using SAM. It primary aims to demonstrate Cloudformation parameters as well as a simplified configuration made possible with SAM Policies, therefore it'll not utilise API Gateway or any other Event source and as a result only the account owner can invoke it.
```
Comment (brettstack) [inline examples/2016-10-31/hello_world_vpc/README.md]: "It primary" => "It primarily"

Comment (brettstack) [reply examples/2016-10-31/hello_world_vpc/README.md]: "Lambda function in VPC" => "Lambda function in a VPC"?

Comment (heitorlessa) [reply examples/2016-10-31/hello_world_vpc/README.md]: done!

**Pull request merged**